### DISCUSSION TITLE
How to make selection.getNodes() get an empty array?

### DISCUSSION THREAD

**Discussion by awehook** [facebook/lexical#2399] [Q&A] [answered]
#1962 
Look at the condition judgment inside the code, selection.getNodes() may get an empty array, under what circumstances will it happen?
https://github.com/facebook/lexical/blob/73ba9caece04bc7138110478e3c89c2bb2d3dcf2/packages/lexical/src/LexicalSelection.ts#L568-L573

**Comment by acywatson** [comment, marked as answer]
It'll happen if only a single empty (no children) element node is selected. One practical case would be the initial editor state, which is just an empty ParagraphNode under the RootNode.
